classes/image_blender.cpp

```diff
@@ -0,0 +1,177 @@
+	ERR_FAIL_COND_MSG(p_src.is_null(), "It's not a reference to a valid Image object.");
+	int dsize = p_dst->get_data().size();
+	int srcdsize = p_src->get_data().size();
+	ERR_FAIL_COND(dsize == 0);
+	ERR_FAIL_COND(srcdsize == 0);
+	ERR_FAIL_COND(p_dst->get_format() != p_src->get_format());
+
+	Rect2i clipped_src_rect = Rect2i(0, 0, p_src->get_width(), p_src->get_height()).clip(p_src_rect);
+
+	if (p_dst_pos.x < 0)
+		clipped_src_rect.position.x = ABS(p_dst_pos.x);
+	if (p_dst_pos.y < 0)
+		clipped_src_rect.position.y = ABS(p_dst_pos.y);
+
+	if (clipped_src_rect.size.x <= 0 || clipped_src_rect.size.y <= 0)
+		return;
+
+	Point2 src_underscan = Point2(MIN(0, p_src_rect.position.x), MIN(0, p_src_rect.position.y));
+	Rect2i dest_rect = Rect2i(0, 0, p_dst->get_width(), p_dst->get_height()).clip(Rect2i(p_dst_pos - src_underscan, clipped_src_rect.size));
+
+	p_dst->lock();
+	Ref<Image> img = p_src;
+	img->lock();
+
+	for (int i = 0; i < dest_rect.size.y; i++) {
+
+		for (int j = 0; j < dest_rect.size.x; j++) {
+
+			int src_x = clipped_src_rect.position.x + j;
+			int src_y = clipped_src_rect.position.y + i;
+
+			int dst_x = dest_rect.position.x + j;
+			int dst_y = dest_rect.position.y + i;
+
+			Color sc = img->get_pixel(src_x, src_y);
+			Color dc = p_dst->get_pixel(dst_x, dst_y);
+
+			p_dst->set_pixel(dst_x, dst_y, blend_colors(sc, dc));
+		}
+	}
+
+	img->unlock();
+	p_dst->unlock();
+}
+
+void ImageBlender::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("blend_rect", "src", "src_rect", "dst", "dst_pos"), &ImageBlender::blend_rect);
+
+	ClassDB::bind_method(D_METHOD("set_rgb_equation", "equation"), &ImageBlender::set_rgb_equation);
+	ClassDB::bind_method(D_METHOD("get_rgb_equation"), &ImageBlender::get_rgb_equation);
+
+	ClassDB::bind_method(D_METHOD("set_alpha_equation", "equation"), &ImageBlender::set_alpha_equation);
+	ClassDB::bind_method(D_METHOD("get_alpha_equation"), &ImageBlender::get_alpha_equation);
+
+	ClassDB::bind_method(D_METHOD("set_rgb_src_factor", "factor"), &ImageBlender::set_rgb_src_factor);
+	ClassDB::bind_method(D_METHOD("get_rgb_src_factor"), &ImageBlender::get_rgb_src_factor);
+
+	ClassDB::bind_method(D_METHOD("set_rgb_src_factor", "factor"), &ImageBlender::set_rgb_dst_factor);
+	ClassDB::bind_method(D_METHOD("get_rgb_src_factor"), &ImageBlender::get_rgb_dst_factor);
+
+	ClassDB::bind_method(D_METHOD("set_rgb_dst_factor", "factor"), &ImageBlender::set_rgb_dst_factor);
+	ClassDB::bind_method(D_METHOD("get_rgb_dst_factor"), &ImageBlender::get_rgb_dst_factor);
+
+	ClassDB::bind_method(D_METHOD("set_alpha_src_factor", "factor"), &ImageBlender::set_alpha_src_factor);
+	ClassDB::bind_method(D_METHOD("get_alpha_src_factor"), &ImageBlender::get_alpha_src_factor);
+
+	ClassDB::bind_method(D_METHOD("set_alpha_dst_factor", "factor"), &ImageBlender::set_alpha_dst_factor);
+	ClassDB::bind_method(D_METHOD("get_alpha_dst_factor"), &ImageBlender::get_alpha_dst_factor);
+
```

Comment:
And yeah, would also make sense to add properties for each of the setter/getter pair, like:

```suggestion
ADD_PROPERTY(PropertyInfo(Variant::INT, "rgb_src_factor"), "set_rgb_src_factor", "get_rgb_src_factor");
```